app/background/wallet/service.js

```diff
@@ -269,42 +328,87 @@ class WalletService {
   };
 
   zap = async () => {
-    this._ensureClient();
+    await this._ensureClient();
     return this.client.zap(WALLET_ID, 'default', 1);
   };
 
   importName = (name, start) => {
     return this._executeRPC('importname', [name, start]);
   };
 
-  rpcGetWalletInfo = () => {
-    return this._executeRPC('getwalletinfo', []);
+  rpcGetWalletInfo = async () => {
+    return await this._executeRPC('getwalletinfo', []);
   };
 
   _onNodeStart = async (networkName, network, apiKey) => {
+    const conn = await getConnection();
+
     this.networkName = networkName;
+    this.apiKey = apiKey;
+    this.network = network;
+
     const walletOptions = {
       network: network,
-      port: network.walletPort,
-      apiKey,
+      port: 12137,
```

Comment:
because this is not the standard wallet port, hs-client will not be able to communicate with Bob's wallet without an extra parameter `--http-port=12137`